Week-9/to-do-app/src/components/containers/TodoContainer.js

```diff
@@ -0,0 +1,280 @@
+      labelVal: "",
+      discriptionVal: "",
+      todos: data,
+      editId: "",
+    };
+  }
+
+  setLabel = (e) => {
+    this.setState({
+      labelVal: e.target.value,
+    });
+  };
+
+  setDescription = (e) => {
+    this.setState({
+      discriptionVal: e.target.value,
+    });
+  };
+
+  addTask = () => {
+    if (this.state.labelVal !== "") {
+      const task = {
+        id: uuidv4(),
+        label: this.state.labelVal,
+        description: this.state.discriptionVal,
+        done: false,
+      };
+
+      const todos = this.state.todos;
+      todos.push(task);
+      this.setState({
+        labelVal: "",
+        discriptionVal: "",
+      });
+
+      toast.success("Task added successfully", {
+        position: "bottom-center",
+        autoClose: 2000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+        theme: "dark",
+      });
+    }
+  };
+
+  taskComplete = (itemId) => {
+    var updatedItems = this.state.todos.map((item) => {
+      if (itemId === item.id) item.done = !item.done;
+
+      return item;
+    });
+
+    this.alertAudio.play();
+
+    // State Updates are Merged
+    this.setState({
+      todos: [].concat(updatedItems),
+    });
+
+    toast.success("Task mark as completed", {
+      position: "bottom-center",
+      autoClose: 2000,
+      hideProgressBar: false,
+      closeOnClick: true,
+      pauseOnHover: true,
+      draggable: true,
+      progress: undefined,
+      theme: "dark",
+    });
+  };
+
+  taskDelete = (itemId) => {
+    var updatedItems = this.state.todos.filter((item) => {
+      return item.id !== itemId;
+    });
+
+    this.setState({
+      todos: [].concat(updatedItems),
+    });
+
+    toast.success("Task deleted sucessfully", {
+      position: "bottom-center",
+      autoClose: 2000,
+      hideProgressBar: false,
+      closeOnClick: true,
+      pauseOnHover: true,
+      draggable: true,
+      progress: undefined,
+      theme: "dark",
+    });
+  };
+
+  taskEdit = (itemId) => {
+    const todoItemIndex = this.state.todos.findIndex(
+      (item) => item.id === itemId
+    );
+    if (todoItemIndex !== -1) {
+      const todo = this.state.todos[todoItemIndex];
+      this.setState({
+        labelVal: todo.label,
+        discriptionVal: todo.description,
+        editId: itemId,
+      });
+    }
+  };
+
+  taskCancel = () => {
+    this.setState({
+      labelVal: "",
+      discriptionVal: "",
+      editId: "",
+    });
+  };
+
+  taskUpdate = () => {
+    let itemId = this.state.editId;
+    const todoItemIndex = this.state.todos.findIndex(
+      (item) => item.id === itemId
+    );
+    if (todoItemIndex !== -1) {
+      const todo = this.state.todos[todoItemIndex];
+      todo.label = this.state.labelVal;
+      todo.description = this.state.discriptionVal;
+
+      this.setState({
+        labelVal: "",
+        discriptionVal: "",
+        editId: "",
```

Comment:
add this.setState({
  labelVal: "",
  discriptionVal: "",
  editId: "",
 todos: this.state.todos
})